### PR TITLE
Zig

### PR DESCRIPTION
**Pull request by yvendruscolo**
Hello! Thank you for your contribution to `polyglot-euler`.
- [x] I checked [CONTRIBUTING.md](https://github.com/FrankKair/polyglot-euler/blob/master/CONTRIBUTING.md) for my programming language of choice.

### Performance

* 1: [Try it online!](https://tio.run/##NY/LCoMwEEX3@YpLoBBpEcW2C1OhH9KN1kcjNRE1ChW/3U5s3QycMzM3mY@q1vVpdD@gH3IkuKumNd0gOCH3JPv1prTT1CTn50VmK98JyRham6HUaFKlhYfRqBwzA8a0Q2@bGPZ6pr1A/l29m8iZ6aXeBUSNG8IgCDzEDo4JQm9LAVTpzAEREkqBoQSiy0b7DNxLbquWGy9UF7h4@qPg8/LQ/ORm6JqFresX "Zig – Try It Online")

* 2: [Try it online!](https://tio.run/##ZY/dCoJAEIXv9ykOQrBSiIV04Sb0IN1ops2aq/gLic9us1YX0cAwzHfmMDNPypflWpm2Q9uliHCmsq6aTjrcOq4Sb22MG8MiMy@9JX3uWaCEQN0nyAzKmIx0MVSUYhLAEDdo@zJEfwzY56sPoy/Zf4n@I8UPGe/0uEFqnBD4a7gIIaeCB0iBuGgFjW2EQs3uuh2gzFo2OCDi7Uz5GDui1cy6TfuAdKb5YpydPZVfncWyvAA "Zig – Try It Online")

* 6: [Try it online!](https://tio.run/##ZZDbCoJAFEXf5ys2PjlWYhA9ZEEfEojmpREdzXEUEr/d5hhS2ONe61xmzktk03SvpGqh2hgXXEVZV01rWyZa3Gcf14eNNNIwN04inbkEfMZQ6wipRBkKaXN0lYgxMKALGyhdBlUaqKcOm0SdoI8HM8Lzv/qP5QvZL@Sns3@IIoGd42y853GcKGxM4PNOrDaSyuEg9xc5E0ojkkKZWcQcMhy7VTeV0Sdtaxhv0trSS8w5RjZNbw "Zig – Try It Online")

### CODE REVIEW

File: src/001/p001.zig [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+pub fn main() void {
+  var sum: u64 = 0;
+  var j: u64 = 3;
+  while (j < 1000) : (j += 1) {
+    if (j % 3 == 0 or j % 5 == 0) {
+      sum += j;
+    }
+  } 
+  warn("{}\n", sum);
+}
```
Comment (FrankKair) [inline src/001/p001.zig]: Missing newline.

File: src/002/p002.zig [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+pub fn main() void {
+  var sum: u64 = 0;
+  var i: u64 = 1;
+  var j: u64 = 1;
+  var k: u64 = 1;
+  while (j < 4000000) : ({k = i; i = j; j += k;}) {
+    if (j % 2 == 0) {sum += j;}
+  }
+  warn("{}\n", sum);
+}
```
Comment (FrankKair) [inline src/002/p002.zig]: Missing newline.

File: src/006/p006.zig [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+pub fn main() void {
+  var sum_of_squares: u64 = 0;
+  var sum: u64 = 0;
+  var j: u64 = 1;
+  var res: u64 = while (j <= 100) : (j += 1) {
+    sum_of_squares += j * j;
+    sum += j;
+  } else (sum * sum) - sum_of_squares;
+  warn("{}\n", res);
+}
```
Comment (FrankKair) [inline src/006/p006.zig]: Missing newline.

File: src/015/p015.clj [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+       (mapv (partial apply +))))
+
+(defn solve [] 
+  (->> init
+      (iterate step)
+      (drop 19)
+      first
+      (reduce ((map #(* % %)) +))))
+
+(prn (solve))
```
Comment (FrankKair) [inline src/015/p015.clj]: Did you originally intend to add a Clojure solution here as well?
(Also, missing newline 😅).

**Comment by yvendruscolo** [comment]
I always forget about the newline...

File: src/015/p015.clj (hunk 4 above)
Comment (yvendruscolo) [reply src/015/p015.clj]: not really...

**Pull request merged**